### PR TITLE
chore(stepfunctions): clean up aws-stepfunctions linter exclusions

### PR DESCRIPTION
**Pull request by shivlaks**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-stepfunctions/lib/fields.ts [hunk 1]
```diff
@@ -39,6 +39,11 @@ export class Data {
     return new JsonPathToken('$').toString();
   }
 
+  /**
+   * Determines if the indicated string is an encoded JSON path
+   *
+   * @param value String to be evaluated
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions/lib/fields.ts]: all lower case 'string'

File: packages/@aws-cdk/aws-stepfunctions/lib/input.ts [hunk 2]
```diff
@@ -53,6 +59,14 @@ export class TaskInput {
...
+  /**
+   * Use a literal string
+   * This might be a JSON-encoded object, or just text.
+   */
   TEXT,
+  /**
+   * Use an object which may contain Data and Context fields
+   * as object values, if desired.
+   */
   OBJECT
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions/lib/input.ts]: Can we provide a few examples to clarify this?

File: packages/@aws-cdk/aws-stepfunctions/lib/state-graph.ts [hunk 3]
```diff
@@ -55,6 +55,10 @@ export class StateGraph {
      */
     private superGraph?: StateGraph;
 
+    /**
+     *
+     * @param startState state that gets executed when the state machine is launched
+     */
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions/lib/state-graph.ts]: document both params.

Also avoid the empty first line inside the comment block. Same in other places.

File: packages/@aws-cdk/aws-stepfunctions/lib/states/wait.ts [hunk 4]
```diff
@@ -4,6 +4,9 @@ import { IChainable, INextable } from '../types';
 import { StateType } from './private/state-type';
 import { State } from './state';
 
+/**
+ * TODO
+ */
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions/lib/states/wait.ts]: ?

File: packages/@aws-cdk/aws-stepfunctions/package.json [hunk 5]
```diff
@@ -88,21 +88,8 @@
...
-      "docs-public-apis:@aws-cdk/aws-stepfunctions.InputType.OBJECT",
-      "docs-public-apis:@aws-cdk/aws-stepfunctions.Data.isJsonPathString",
-      "docs-public-apis:@aws-cdk/aws-stepfunctions.Result.value",
-      "docs-public-apis:@aws-cdk/aws-stepfunctions.StateGraph.startState",
-      "docs-public-apis:@aws-cdk/aws-stepfunctions.TaskInput.type",
-      "docs-public-apis:@aws-cdk/aws-stepfunctions.InputType.TEXT",
-      "docs-public-apis:@aws-cdk/aws-stepfunctions.WaitTime",
-      "docs-public-apis:@aws-cdk/aws-stepfunctions.ActivityProps",
-      "docs-public-apis:@aws-cdk/aws-stepfunctions.IActivity"
+      "no-unused-type:@aws-cdk/aws-stepfunctions.ServiceIntegrationPattern"
```
Comment (nija-at) [inline packages/@aws-cdk/aws-stepfunctions/package.json]: (Not related to this PR)

I'm wondering - if this class isn't used anywhere in this package, should it instead be moved to the stepfunction-tasks module?

Comment (shivlaks) [reply packages/@aws-cdk/aws-stepfunctions/package.json]: that was my thinking as well. I'm intending to include that in another PR (there's a few other things that need to be cleaned up between these 2 packages)

Comment (nija-at) [reply packages/@aws-cdk/aws-stepfunctions/package.json]: > there's a few other things

can we capture this 'list' somewhere? I would suggest opening an issue, listing it down and linking it off StepFunctions tracking issue.

File: packages/@aws-cdk/aws-stepfunctions/lib/input.ts (hunk 2 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-stepfunctions/lib/input.ts]: added

File: packages/@aws-cdk/aws-stepfunctions/lib/state-graph.ts (hunk 3 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-stepfunctions/lib/state-graph.ts]: done

File: packages/@aws-cdk/aws-stepfunctions/lib/fields.ts (hunk 1 above)
Comment (shivlaks) [reply packages/@aws-cdk/aws-stepfunctions/lib/fields.ts]: done

**Pull request merged**